rotkehlchen/chain/evm/decoding/monerium/decoder.py

```diff
@@ -0,0 +1,110 @@
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class MoneriumCommonDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            native_asset: Asset,
+            asset_address: ChecksumEvmAddress,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.native_asset = native_asset
+        self.asset_address = asset_address
+
+    def _decode_monerium(self, context: DecoderContext) -> DecodingOutput:
+        """Decode Monerium when it is minted or burned"""
+
+        if context.tx_log.topics[0] != MIGRATED:
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(context.all_logs[0].topics[1])
+        to_address = hex_or_bytes_to_address(context.all_logs[0].topics[2])
+
+        if not self.base.is_tracked(from_address) and not self.base.is_tracked(to_address):
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data)
+        amount = token_normalized_value_decimals(amount_raw, 18)
+
+        event = None
+        if from_address == '0x0000000000000000000000000000000000000000':
+            # Create a mint event
+            mint_event = self.base.make_event_from_transaction(
+                transaction=context.transaction,
+                tx_log=context.tx_log,
+                event_type=HistoryEventType.RECEIVE,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=self.native_asset,
+                balance=Balance(amount=amount),
+                location_label=to_address,
+                notes=f'Mint {amount} EURe',
+                counterparty=CPT_MONERIUM,
+                address=self.asset_address,
+            )
+            event = mint_event
+
+        elif (to_address == '0x0000000000000000000000000000000000000000' and
+                context.transaction.input_data.hex().startswith('3823caec')):
+            # Create a burn event
+            burn_event = self.base.make_event_from_transaction(
+                transaction=context.transaction,
+                tx_log=context.tx_log,
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=self.native_asset,
+                balance=Balance(amount=amount),
+                location_label=from_address,
+                notes=f'Burn {amount} EURe',
+                counterparty=CPT_MONERIUM,
+                address=self.asset_address,
+            )
+            event = burn_event
+
+        else:
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
You don't need the else since the event variable is set to None by default and then you have the same as `DEFAULT_DECODING_OUTPUT` if you return in 101